rosbot_gazebo/test/test_utils.py

```diff
@@ -14,94 +14,103 @@
-        self.set_parameters([use_sim_time])
+        # Robot callback
+        self.joint_sub = self.create_subscription(
+            JointState, "joint_states", self.joint_states_callback, 10
+        )
+        self.controller_sub = self.create_subscription(
+            Odometry, "rosbot_base_controller/odom", self.controller_callback, 10
+        )
+        self.ekf_sub = self.create_subscription(
+            Odometry, "odometry/filtered", self.ekf_callback, 10
+        )
+        self.imu_sub = self.create_subscription(Imu, "imu_broadcaster/imu", self.imu_callback, 10)
 
-        self.VELOCITY_STABILIZATION_DELAY = 3
-        self.current_time = 1e-9 * self.get_clock().now().nanoseconds
-        self.goal_received_time = 1e-9 * self.get_clock().now().nanoseconds
-        self.vel_stabilization_time_event = Event()
+        # Sensor callback
+        self.camera_rgb_sub = self.create_subscription(
+            Image, "camera/color/image_raw", self.camera_image_callback, 10
+        )
+        self.camera_pc_sub = self.create_subscription(
+            PointCloud2, "camera/depth/points", self.camera_points_callback, 10
+        )
+        self.range_subs = []
+        for range_topic_name in self.RANGE_SENSORS_TOPICS:
+            sub = self.create_subscription(LaserScan, range_topic_name, self.ranges_callback, 10)
+            self.range_subs.append(sub)
+        self.scan_sub = self.create_subscription(LaserScan, "scan", self.scan_callback, 10)
+
+        # Timer - send cmd_vel and check if the time needed for speed stabilization has elapsed
+        self.timer = self.create_timer(0.1, self.timer_callback)
 
+        # Destination velocity for cmd_vel
         self.v_x = 0.0
         self.v_y = 0.0
         self.v_yaw = 0.0
-        self.twist = None  # Debug info
 
-        self.controller_odom_flag = False
-        self.ekf_odom_flag = False
-        self.odom_tf_event = Event()
-        self.scan_event = Event()
-        self.ranges_events = [Event() for _ in range(len(self.RANGE_SENSORS_TOPICS))]
+        # Debug values
+        self.controller_twist = None
+        self.ekf_twist = None
+
+        # Robot test flags and event
+        self.is_controller_msg = False
+        self.is_controller_odom_correct = False
+        self.is_ekf_msg = False
+        self.is_ekf_odom_correct = False
+        self.is_imu_msg = False
+        self.is_joint_msg = False
+        self.robot_initialized_event = Event()
+        self.vel_stabilization_time_event = Event()
+
+        # Sensor test events
         self.camera_color_event = Event()
         self.camera_points_event = Event()
+        self.ranges_events = [Event() for _ in range(len(self.RANGE_SENSORS_TOPICS))]
+        self.scan_event = Event()
 
-    def clear_odom_flag(self):
-        self.controller_odom_flag = False
-        self.ekf_odom_flag = False
+        # Parameters Use simulation time to correct run on slow machine
```

Comment:
I think the comment is not right. Using `use_sim_time` parameter changes the  clock source from POSIX time to time published at the `/clock` topic.
```suggestion
        # Using /clock topic as time source
```